pkg/function/kube_task.go

```diff
@@ -64,6 +65,18 @@ func kubeTask(ctx context.Context, cli kubernetes.Interface, namespace, image st
 		Command:      command,
 		PodOverride:  podOverride,
 	}
+	// Mark labels to pods with prefix `kanister.io`. Add the jobID as reference to the origin for the pod.
+	if fields := field.FromContext(ctx); fields != nil {
+		for _, f := range fields.Fields() {
+			if strings.HasPrefix(f.Key(), consts.LabelPrefix) {
+				jobID := f.Value().(string)
+				if options.Labels == nil {
+					options.Labels = make(map[string]string)
+				}
+				options.Labels[consts.LabelPrefix+"JobID"] = jobID
+			}
+		}
+	}
```

Comment:
is exporting this to another util helpful? my understanding is we would use this in other kanister functions as well.

Reply:
Yes, very true. I will move this.